src/index.ts

```diff
@@ -120,5 +141,17 @@ function parseLineHeight(value: string) {
 	if (parsed.toString() === value) {
 		return parsed;
 	}
+	const match = /^(\+)?(\.)?/.exec(value);
```

Comment:
OK the more I look at this logic, the more I think installing [`parse-css-dimension`](https://github.com/jedmao/parse-css-dimension) might be useful in replacing this logic. What do you think?

Reply:
No broken tests with the following implementation:
```ts
function parseLineHeight(value: string) {
	try {
		const parsed = parseCssDimension(value);
		if (parsed.type === 'number') {
			return parsed.value;
		}
	} catch (err) {
		// noop
	}
	return value;
}
```

Reply:
yes that would probably be better -- ideally the parsing should also be extended to reject negative values for line height (and invalid dimensions/units)

Reply:
I see what you mean. Looks like there's a validation aspect that needs to be employed here. I just tried this:
```css
.foo {
  line-height: -1;
}
```
on http://jigsaw.w3.org/css-validator/#validate_by_input and got the following error:
> Value Error : line-height -1 negative values are not allowed
We could easily add the following:
```js
if (parsed.value < 0) {
    throw error(`Value Error: line-height: ${parsed.value}; negative values are not allowed.`);
}
```